### ISSUE TITLE
The `project` swift version is not populated by cocoapods to the targets

### ISSUE THREAD

**Issue by zbencz3** [CocoaPods/CocoaPods#7613]
🌈
* [X] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

pod repo update
pod install

## What did you expect to happen?

Since my `project` version is set to `SWIFT_VERSION` 4.1, I expected so the `SWIFT_VERSION` of the targets I have to be set to 4.1 also.

<img width="1069" alt="project" src="https://user-images.githubusercontent.com/5381246/38556681-559082ce-3c7f-11e8-9e89-d6d70c4a5118.png">

<img width="1023" alt="target" src="https://user-images.githubusercontent.com/5381246/38556736-905f0cea-3c7f-11e8-8d5c-90cee90cd4d8.png">

## What happened instead?

Instead the value set is:
`SWIFT_VERSION = 3.3`
and the the `Pods-XXX.debug.xcconfig` and `Pods-XXX.release.xcconfig` files also have a version 3.0 set.
<img width="221" alt="screen shot 2018-04-10 at 5 22 11 am" src="https://user-images.githubusercontent.com/5381246/38556622-26178e2a-3c7f-11e8-8172-1209dea7f921.png">

### Stack

```
   CocoaPods : 1.5.0
        Ruby : ruby 2.3.3p222 (2016-11-21 revision 56859) [universal.x86_64-darwin17]
    RubyGems : 2.5.2
        Host : Mac OS X 10.13.3 (17D102)
       Xcode : 9.3 (9E145)
         Git : git version 2.15.1 (Apple Git-101)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 0219b182a76035e826f14b110b0ea0638c3cb4f4
               MyAppURL - https://github.com/MyAppURL/Specs.git @ e463950fd18783222495cf7fcd8ae71c706a944e
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
source 'https://github.com/MyAppURL/Specs.git'
source 'https://github.com/CocoaPods/Specs.git'

use_frameworks!
inhibit_all_warnings!

project 'My. App.xcodeproj'

def set_ios_platform
  platform :ios, '10.3'
end

def set_osx_platform
  platform :osx, '10.10'
end

def import_core_pods
  pod "SAMKeychain", '~> 1.5'
  pod "PLManagedModel", '0.3.1'
  pod "ReactiveSwift", '~> 1.1.5'
  pod "ReactiveObjC", '~> 2.1.2'
  pod "ReactiveCocoa", '~> 5.0.0'
  pod "ReactiveObjCBridge", '~> 1.0.0'
  pod "DateTools", '~> 2.0.0'
  pod "DateToolsSwift", '~> 2.0.0'
  pod "CocoaLumberjack/Swift", '~> 3.0.0'
  pod "SSZipArchive", '~> 1.1'
end

def import_debug_pods
  pod "Aspects", configurations: ['Debug']
  pod "Reveal-SDK", configurations: ['Debug']
end

def import_ios_pods
  import_core_pods
  import_debug_pods
  pod "AFNetworking", '~> 2.4.1'
  pod "SVProgressHUD", '~> 1.1.2'
  pod "TPKeyboardAvoiding", '~> 1.2.6'
  pod "UIView-Autolayout", '~> 0.2.0'
  pod "JDStatusBarNotification", '~> 1.5.5'
  pod "WYPopoverController", git: "https://github.com/sammcewan/WYPopoverController.git"
  pod "RXCollections", '~> 1.0'
  pod "MGSwipeTableCell", '~> 1.1.1'
  pod "Mixpanel", '~> 3.2'
  pod "StaticDataTableViewController", '~> 2.0.3'
  pod "MCSwipeTableViewCell", '~> 2.1.2'
  pod "Intercom", '~> 3.0'
  pod "DZNEmptyDataSet", '~> 1.7.2'
  pod "XLForm", git: 'https://github.com/xxx/XLForm', commit: 'f6a30eb6c74ffff6286f491e9a360c408d06b597'
  pod "JVFloatLabeledTextField", '~> 1.1.0'
  pod "Fabric"
  pod "Crashlytics"
  pod "Dwifft", '~> 0.5.0'
  pod "MMMarkdown", '~> 0.5.5'
end

def import_test_pods
  pod "OCMock", '3.2.2'
  pod "Foundry", git: "https://github.com/xxx/Foundry", branch: 'master'
  pod "Quick", '~> 1.1.0'
  pod "Nimble", '~> 7.0.2'
end

target "My App" do
  set_ios_platform
  import_ios_pods
end

target "My App Dev" do
  set_ios_platform
  import_ios_pods

  target "My. AppTests" do
    inherit! :search_paths
    import_test_pods
  end
end

target "SeedBuilder" do
  set_osx_platform
  import_core_pods

  target "SeedBuilderTests" do
    inherit! :search_paths
    import_test_pods
  end
end

post_install do |installer|
  installer.pods_project.build_configurations.each do |config|
    config.build_settings['SWIFT_VERSION'] = '4.1'
  end
  
  installer.pods_project.targets.each do |target|
    if ['Pods-My App', 'Pods-My App Dev', 'Pods-My. AppTests', 'Pods-SeedBuilder', 'Pods-SeedBuilderTests'].include? target.name
      target.build_configurations.each do |config|
        config.build_settings['SWIFT_VERSION'] = '4.1'
      end
    else
      target.build_configurations.each do |config|
        config.build_settings['SWIFT_VERSION'] = '3.2'
      end
    end
  end
#  installer.pods_project.add_build_configuration('DEBUG_INFORMATION_FORMAT', 'dwarf-with-dsym')
end
```

**Comment by dnkoutso** [comment]
need a sample app to debug, also when you say "targrets" which targets do you mean? If you mean the ones within your `.xcodeproj` then you set the `SWIFT_VERSION` manually there.

**Comment by zbencz3** [comment]
thanks @dnkoutso. 
I mean the ones within my xcodproject as you said.
<img width="176" alt="targets" src="https://user-images.githubusercontent.com/5381246/38567158-278326ba-3cee-11e8-87e2-8c4f0685bd4d.png">


If I set the `SWIFT_VERSION` explicitly there, then I will have the warning 
```
[!] The `My App Dev [Debug]` target overrides the `SWIFT_VERSION` build setting defined in `Pods/Target Support Files/Pods-My App Dev/Pods-My App Dev.debug.xcconfig'. This can lead to problems with the CocoaPods installation
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.
```

I wanted to set things up without having this warning. Is that possible?

**Comment by dnkoutso** [comment]
hmmm this is a bit weird, would love to see a quick sample app to repro as a quick scan in the code does not show `SWIFT_VERSION` being set in your project.

**Comment by dnkoutso** [comment]
@zbencz3 does any of the pods you consume actually set `'SWIFT_VERSION` in `s.xcconfig` or `s.user_xcconfig`? Perhaps one of them is overriding this setting because I know the default CocoaPods code does not.

**Comment by zbencz3** [comment]
will have to check them one by one. Will let you know asap

**Comment by dnkoutso** [comment]
@zbencz3 I used the same pods as you and debugged some of the code, as I suspected this was the problem.

https://github.com/CocoaPods/Specs/blob/d0ec5a65e80656c8d78e12ff19f251df879e0bc2/Specs/d/1/8/CocoaLumberjack/3.0.0/CocoaLumberjack.podspec.json#L75

Going to close at the moment as this is not a library issue, we *could* however skip explicitly this value from being integrated.

Pod authors should *always* use `s.pod_target_xcconfig` and not mess with the users settings.

My gut says that Xcode defaults to `3.3` when you set the value `3.0`.
